test/users/index.spec.js

```diff
@@ -0,0 +1,52 @@
+const request = require('supertest');
+
+const app = require('../../app');
+const { createManyUsers } = require('../factory/users_factory');
+const { showUserSerializer } = require('../../app/serializers/users');
+
+describe('GET /users', () => {
+  const httpRequest = (query = {}) =>
+    request(app)
+      .get('/users')
+      .query(query);
+  let usersIndexResponse = {};
+  let users = {};
+  beforeAll(async () => {
+    users = await createManyUsers(2);
+  });
+
+  describe('without query params', () => {
+    beforeAll(async () => {
+      usersIndexResponse = await httpRequest();
+    });
+
+    test('Responds with 200 status code', () => expect(usersIndexResponse.statusCode).toBe(200));
+
+    test('Responds with the expected body structure', () =>
+      expect(usersIndexResponse.body).toMatchObject({
+        page: expect.any(Array),
+        total_count: expect.any(Number)
+      }));
+
+    test('Responds with the expected users', () => {
+      expect(usersIndexResponse.body.page).toEqual(users.map(showUserSerializer));
+    });
+  });
+
+  describe('with query params', () => {
+    beforeAll(async () => {
+      usersIndexResponse = await httpRequest({ page: 2, limit: 1 });
+    });
+
+    test('Responds with 200 status code', () => expect(usersIndexResponse.statusCode).toBe(200));
+
+    test('Responds with the expected body structure', () =>
+      expect(usersIndexResponse.body).toMatchObject({
+        page: expect.any(Array),
+        total_count: expect.any(Number)
+      }));
+    test('Responds with the expected users', () => {
+      expect(usersIndexResponse.body.page).toEqual([showUserSerializer(users[1])]);
+    });
+  });
+});
```

Comment:
There are paths that are not being tested:
- What if page is a string?
- What if limit is a string?
- What if both are strings?
- What if the user is not authenticated?
- What if I ask for a page with no users?

Reply:
1, 2, 3 --> field validation error
4 --> 401
5 --> empty page